### PR TITLE
Update XDG files

### PR DESCRIPTION
**Pull request by scx**
Update AppData file:
 - Fix `description`
 - Fix `screenshot`
 - Remove `categories` (should be obtained from desktop file)
 - Update `homepage` URL
 - Add `faq` URL
 - Add `donation` URL
 - Add `translate` URL

See:
https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-url

Update desktop file:
 - Add `Version` entry
 - Add `TryExec` entry
 - Add `StartupWMClass` entry
 - Add `Keywords` entries
 - Fix `GenericName` (should be: `Text Editor`, `Advanced Text Editor` or `Source Code Editor`)
 - Add `X-GNOME-FullName` entries

See:
https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#id1341

---

See: https://github.com/notepadqq/notepadqq/pull/896

### CODE REVIEW

**Comment by scx** [comment]
It failed because of this issue:
https://github.com/flathub/org.freedesktop.appstream-glib/issues/6

However, my patch is already merged:
https://github.com/flathub/org.freedesktop.appstream-glib/pull/7

**Comment by danieleds** [comment]
Can we upgrade this to runtime-version 5.12 or newer?

**Comment by scx** [comment]
> Can we upgrade this to runtime-version 5.12 or newer?

This PR is about the current version. It is based on QtWebKit. Unfortunately, QtWebKit for KDE 5.12 doesn't support all arches, so we stick with 5.11.
However, it shouldn't be a problem, as we will migrate to QtWebEngine soon. It will use KDE 5.12 (or 5.13, if you really want to - although I think that LTS version is a better choice). 

I will setup a beta branch for 1.9.99 soon.

**Comment by scx** [comment]
bot, build com.notepadqq.Notepadqq

**Comment by scx** [comment]
It seems that `appstream-util` is still outdated in Flathub Infra. :angry:

**Comment by scx** [comment]
bot, build com.notepadqq.Notepadqq

File: notepadqq-1.4.8-xdg.patch [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+diff --git a/support_files/notepadqq.appdata.xml b/support_files/notepadqq.appdata.xml
+index 5f0d67552..e00dcb8e5 100644
+--- a/support_files/notepadqq.appdata.xml
++++ b/support_files/notepadqq.appdata.xml
+@@ -28,6 +28,9 @@
+   <url type="homepage">https://github.com/notepadqq/notepadqq</url>
+   <url type="bugtracker">https://github.com/notepadqq/notepadqq/issues</url>
+   <url type="help">https://github.com/notepadqq/notepadqq/wiki</url>
++  <content_rating type="oars-1.0">
++    <content_attribute id="social-info">moderate</content_attribute>
```
Comment (TingPing) [inline notepadqq-1.4.8-xdg.patch:86]: Curious what that would be?

Comment (scx) [reply notepadqq-1.4.8-xdg.patch:86]: @TingPing 
> Curious what that would be?

Optional (disabled by default) collecting anonymous statistics.
https://github.com/notepadqq/notepadqq/blob/v1.4.8/src/ui/frmpreferences.ui#L115-L119
> Collect and transmit anonymous statistics to improve Notepadqq

You already know that. We've talked about this.

This is not a new entry. It was already available in the previous patch.
https://github.com/flathub/com.notepadqq.Notepadqq/pull/2/files#diff-b62249923f820079744a0bdf13d405efL31

**Pull request merged**